packages/emnapi/src/value/create.ts

```diff
@@ -284,15 +284,38 @@ function napi_create_buffer (
 // @ts-expect-error
 ): napi_status {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  let value: number
+  let value: number, pointer: number
 
   $PREAMBLE!(env, (envObject) => {
     $CHECK_ARG!(envObject, result)
-    $from64('result')
-    const arrayBuffer = createArrayBuffer(size, data)
-    const buffer = Buffer.from(arrayBuffer)
+
+    pointer = $makeMalloc('napi_create_buffer', 'size')
+    if (!pointer) throw new Error('Out of memory')
+    $from64('size')
+    for (let i = pointer; i < pointer + size; ++i) {
```

Comment:
`HEAPU8.subarray(pointer, pointer + size).fill(0)` ?

In theory that should be cheaper as engines can use SIMD-optimized `memset` directly.